snuba/subscriptions/data.py

```diff
@@ -230,18 +272,20 @@ def add_conditions(
             ),
         ]
 
```

Comment:
You'll need to add an organization condition here as well.

Reply:
> I wouldn't bother with updating anything Legacy related. It is not used in prod at all and the only reason it's not outright deleted is to support onprem/oss that are not migrated yet

There is initial validation that will fail if I don't update the legacy stuff

Reply:
Extensions are deprecated. The `SnQLSubscriptionData` needs to check for an organization ID and add a condition for it, in the same way that it adds a project_id.